src/box/iproto_constants.h

```diff
@@ -57,8 +57,13 @@ enum {
 	_(COMMIT, 0)							\
 	/** Set for the last row of a tx residing in limbo. */		\
 	_(WAIT_SYNC, 1)							\
-	/** Set for the last row of a synchronous tx. */		\
+	/** Set for the last row of a tx waiting for confirmation. */	\
 	_(WAIT_ACK, 2)							\
+	/**
+	 * Set for the last row of a tx that is immediately acknowledged to the
+	 * user.
+	 */	                                                        \
+	_(EARLY_ACK, 3)							\
```

Comment:
I don't understand. Txns are either sync or async. "Early ack" is not a transaction type. It is behaviour which is deductible from the first txn in the limbo and the split brain handling mode. Why is it made a new txn type?

Reply:
> It is behaviour which is deductible from the first txn in the limbo and the split brain handling mode.

We want replicas to follow the limbo owner in terms of asynchronous transaction processing. 

This is also needed for recovery. I.e., we could have replicated an asynchronous transaction as confirmable one, so after recovery we need to either keep in the limbo, or confirm and evict it.